pkg/crdtest/main_test.go

```diff
@@ -0,0 +1,221 @@
+		return
+	}
+
+	if *prettyWatch {
+		fmt.Println("---")
+		fmt.Println(string(b))
+	}
+
+	klog.Infof("Applying %s/%s", obj.GetNamespace(), obj.GetName())
+	if err := globals.k8sClient.Patch(context.Background(), obj, client.Apply, client.FieldOwner("crdtest")); err != nil {
+		if *prettyWatch {
+			fmt.Printf("❌ %s: %v\n", path, err)
+		}
+		klog.Errorf("Failed to apply object from %s: %v", path, err)
+	} else {
+		if *prettyWatch {
+			fmt.Printf("✅ %s\n", path)
+		}
+		klog.Infof("Successfully applied object from %s", path)
+	}
+	fmt.Println()
+}
+
+func watchAndTest(watchDir string) {
+	klog.Info("Watching for changes... (press CTRL-c to exit)")
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		klog.Fatalf("Failed to create watcher: %v", err)
+	}
+	defer watcher.Close()
+
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Create == fsnotify.Create {
+					if filepath.Ext(event.Name) == ".yaml" || filepath.Ext(event.Name) == ".yml" {
+						runTestOnFile(event.Name)
+					}
+				}
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				klog.Errorf("Watcher error: %v", err)
+			}
+		}
+	}()
+
+	err = watcher.Add(watchDir)
+	if err != nil {
+		klog.Fatalf("Failed to add dir to watcher: %v", err)
+	}
+
+	// Run tests on existing files
+	files, err := os.ReadDir(watchDir)
+	if err != nil {
+		klog.Fatalf("Failed to read dir %s: %v", watchDir, err)
+	}
+	for _, file := range files {
+		if !file.IsDir() && (filepath.Ext(file.Name()) == ".yaml" || filepath.Ext(file.Name()) == ".yml") {
+			runTestOnFile(filepath.Join(watchDir, file.Name()))
+		}
+	}
+
+	klog.Infof("Watching for changes in %s...", watchDir)
+	select {} // Block forever
+}
+
+func TestMain(m *testing.M) {
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	klog.Info("Running crdtest")
+
+	scheme := runtime.NewScheme()
+
+	var (
+		restConfig *rest.Config
+		testEnv    *envtest.Environment
+		err        error
+	)
+
+	corev1.AddToScheme(scheme)
+	klog.Info("Added corev1 to scheme")
+
+	v1alpha2.Install(scheme)
+	klog.Info("Added v1alpha2 to scheme")
+
+	if *kubeConfig != "" {
+		klog.Infof("Using kubeConfig=%q", *kubeConfig)
+		restConfig, err = clientcmd.BuildConfigFromFlags("", *kubeConfig)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to get restConfig from BuildConfigFromFlags: %v", err))
+		}
+	} else {
+		klog.Info("Using testenv")
+
+		// The version used here MUST reflect the available versions at
+		// controller-runtime repo: https://raw.githubusercontent.com/kubernetes-sigs/controller-tools/HEAD/envtest-releases.yaml
+		// If the envvar is not passed, the latest GA will be used
+		k8sVersion := os.Getenv("K8S_VERSION")
+
+		var paths []string
+		if *crdDir != "" {
+			paths = []string{*crdDir}
+		} else {
+			paths = []string{
+				filepath.Join("..", "..", "config", "crd", "standard"),
```

Comment:
I think correct path is 
`filepath.Join("config", "crd", "standard")`, then you also don't need to pass `-crdDir config/crd/standard`

Reply:
You need to do the ../.. so `go test ./pkg/crdtest` will work. The unit tests run with the working directory.